### DISCUSSION TITLE
Use values from Pinia store to build useFetch request

### DISCUSSION THREAD

**Discussion by konradsienkowski** [nuxt/nuxt#17440] [General]
Hey there,

In my Nuxt3 application, I need to use values from the Pinia store inside the API request path. For now, I'm using computed property to build it:

```
const config = useRuntimeConfig()
const options = useOptionsStore()
const route = useRoute()

// Define request path
const path = computed(() => {
  return 'contents/docs/articles/' + route.params.category + '/' + route.params.filename + '.md?ref=' + options.versions.active
})

// Fetch data from API
const { data: markdown, refresh } = await useFetch(path.value, { 
  baseURL: config.API_BASE_URL, pick: ['content']
})

// Refresh on version change
options.$onAction(function ({ name, after }) {
  after(() => {
    if (name === 'setActive') {
      console.log(path.value)
      refresh()
    }
  })
})
```

That solution works great, but only for the first fetch. My plan was to subscribe to Pinia actions and use the `refresh` method every time an action called 'setActive' is performed. 

When I'm changing the active version (using a switch located in another component):
- path value in console.log is correctly updated
- refresh method is using the old request path

I've also tried to use the [watch option](https://github.com/nuxt/framework/issues/4970#issuecomment-1125907379) inside useFetch, but it didn't affect the problem. 

Does anybody have any ideas/solutions for that problem? Thanks in advance!

**Comment by farin** [comment]
what about using callback for useFetch ( like https://v3.nuxtjs.org/guide/features/data-fetching/#refreshing-data )
and store as refs to watch changes

something like following
```
const { versions}  = storeToRefs(useOptionsStore())

const { data: markdown, refresh } = await useFetch(() => {
  const { category, filename } = route.params
  return `contents/docs/articles/${category}/${filename}.md?ref=${options.versions.active}`
}, { 
  baseURL: config.API_BASE_URL, pick: ['content']
})

watch(versions, refresh}
```

**Reply by konradsienkowski** [reply]
@farin Thanks! storeToRefs helped me solve that issue
